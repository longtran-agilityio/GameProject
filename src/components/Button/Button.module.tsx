// libs
import { styled, Button } from '@mui/material'

export const StyledButton = styled(Button)(({ theme }) => ({
  display: 'flex',
  color: theme.palette.primary.main,
  textTransform: 'capitalize',
  fontWeight: 600,
  padding: theme.spacing(0),
  disableTransition: {
    transition: 'none',
  },
  minWidth: theme.spacing(0),
  '&:hover': {
    color: theme.palette.text.secondary,
    backgroundColor: 'transparent',
    boxShadow: 'none',
  },
  '& .Mui-focusVisible': {
    backgroundColor: 'transparent',
  },
}))
