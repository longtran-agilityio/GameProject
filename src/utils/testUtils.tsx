import { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { ThemeProvider } from '@mui/material'
import { theme } from '@webapp/themes/index'

const AllBrowserRouter = ({ children }: { children: React.ReactNode }) => (
  <BrowserRouter>
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  </BrowserRouter>
)

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllBrowserRouter, ...options })

export * from '@testing-library/react'
export { customRender as render }
