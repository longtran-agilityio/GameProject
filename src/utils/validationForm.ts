export const EmailValidation = {
  required: 'Email is required',
  pattern: {
    value: /^[\w-.]+@([\w-]+\.)+[\w-]{2,4}$/g,
    message: 'Invalid Email',
  },
}

export const PasswordValidation = {
  required: 'Password is required',
  minLength: { value: 8, message: 'min length is 8' },
}

export const FieldEmpty = {
  required: 'This field is required',
}
