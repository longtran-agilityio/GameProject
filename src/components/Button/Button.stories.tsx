// libs
import { ThemeProvider } from '@mui/material'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import AddIcon from '@mui/icons-material/Add'
import DoneIcon from '@mui/icons-material/Done'
import LocalMallOutlinedIcon from '@mui/icons-material/LocalMallOutlined'
import CancelIcon from '@mui/icons-material/Cancel'
import ArrowRightAltIcon from '@mui/icons-material/ArrowRightAlt'

// component
import { StyledButton } from './Button.module'
import { theme } from '@webapp/themes/index'

export default {
  title: 'Components/Button',
  component: StyledButton,
} as ComponentMeta<typeof StyledButton>

const Template: ComponentStory<typeof StyledButton> = (args) => (
  <ThemeProvider theme={theme}>
    <StyledButton {...args} />
  </ThemeProvider>
)

export const AddButton = Template.bind({})
AddButton.args = {
  children: 'Add to cart',
  size: 'small',
  endIcon: <AddIcon />,
  disableRipple: true,
}

export const CompletedButton = Template.bind({})
CompletedButton.args = {
  children: 'Added',
  size: 'small',
  endIcon: <DoneIcon />,
  disableRipple: true,
  disabled: true,
}

export const CartButton = Template.bind({})
CartButton.args = {
  children: 'Cart',
  size: 'small',
  startIcon: <LocalMallOutlinedIcon />,
  disableRipple: true,
}

export const ClearButton = Template.bind({})
ClearButton.args = {
  size: 'small',
  startIcon: <CancelIcon />,
  disableRipple: true,
}

export const CheckoutButton = Template.bind({})
CheckoutButton.args = {
  children: 'Checkout',
  size: 'small',
  endIcon: <ArrowRightAltIcon />,
  disableRipple: true,
}
