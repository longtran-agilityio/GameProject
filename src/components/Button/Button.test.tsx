import { render, fireEvent } from '@webapp/utils/testUtils'
import { faker } from '@faker-js/faker'

import { StyledButton } from './Button.module'

const onClick = jest.fn()
const ButtonProps = {
  children: faker.name.fullName(),
}
const ButtonComponent = () => {
  return render(<StyledButton onClick={onClick}>{ButtonProps.children}</StyledButton>)
}

describe('Button', () => {
  test('should matches snapshot', () => {
    const { container } = ButtonComponent()
    expect(container).toMatchSnapshot()
  })

  test('It should render correctly', () => {
    const { getByText } = ButtonComponent()
    expect(getByText(ButtonProps.children)).toBeInTheDocument()
  })

  test('On click function should be called once when click button', () => {
    const { getByText } = ButtonComponent()
    fireEvent.click(getByText(ButtonProps.children))
    expect(onClick).toHaveBeenCalledTimes(1)
  })
  test('Button should be enabled', () => {
    const { getByText } = ButtonComponent()
    expect(getByText(ButtonProps.children)).toBeEnabled()
  })
})
