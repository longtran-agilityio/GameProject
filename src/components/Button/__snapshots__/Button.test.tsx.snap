// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should matches snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-17rxn03-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Heather Wisozk
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
